### ISSUE TITLE
[Bug]: first launch of odometer-style SIMT kernel loses all programs with pid>=group_tail — output cells never written (deterministic 30/50 stale); same-binary relaunch complete

### ISSUE THREAD

**Issue by gavinwang269** [triton-lang/triton-ascend#2292]
### Describe the bug

An Inductor(`triton_experimental` backend)-compiled odometer-style SIMT reduction kernel **deterministically loses all programs with `pid >= group_tail` on its FIRST launch** after compilation: the output cells those programs would write are never written (stale), while all programs with `pid < group_tail` land correctly. Re-launching the **same binary in the same process** writes the full output (no error, no log). It is a launch/landing loss ("tiles unwritten"), not a math miscompile — the kernel's ttir is correct line-by-line.

Downstream symptom: in torch inductor's NPU opinfo runs with `TORCHINDUCTOR_NPU_BACKEND=triton_experimental`, the broadcast-backward reduction family shows **35 deterministic f16/f32 wrong-value rows** (add/atan2/clamp/clamp_max/clamp_min/copysign/fmax/fmin/hypot/ldexp/max_binary/maximum/min_binary/minimum/nextafter/pow/special_xlog1py/xlogy/where): `Mismatched elements: 30 / 50 (60.0%)` — exactly the cells owned by the high-pid programs. Even `aten.add`'s broadcast gradient computes wrong, so the defect is op- and dtype-independent.

Environment: Ascend 910B4 / CANN 9.1.0 / triton-ascend 3.6.0 / torch 2.14.0a0+git / torch_npu 2.14.0+gitb4c7212, only via `TORCHINDUCTOR_NPU_BACKEND=triton_experimental`.

### Minimal complete example

Run on an idle 910B in a fresh single process (no pytest, no probes, no coexistence needed; reproduced 5/5 across ops and seeds):

```python
import os
os.environ["TORCHINDUCTOR_NPU_BACKEND"] = "triton_experimental"
os.environ["TORCHINDUCTOR_FORCE_DISABLE_CACHES"] = "1"
import torch, torch_npu
import torch._inductor.config as icfg
icfg.compile_threads = 0; icfg.implicit_fallbacks = False
torch.manual_seed(7)
x = torch.randn(10, 5, device='npu').reshape(50).as_strided((10, 1, 5), (5, 5, 1))
y = torch.randn(10, 5, device='npu')
a = x.clone().detach().requires_grad_(True); b = y.clone().detach().requires_grad_(True)
g = torch.autograd.grad(torch.compile(torch.add)(a, b).sum(), (a, b)); torch.npu.synchronize()
a2 = x.clone().detach().requires_grad_(True); b2 = y.clone().detach().requires_grad_(True)
r = torch.autograd.grad((a2 + b2).sum(), (a2, b2)); torch.npu.synchronize()
for i, (p, e) in enumerate(zip(g, r)):
    big = (p - e).abs() > 1e-4
    print(f"grad{i} n_big={int(big.sum())} idx={big.flatten().nonzero().flatten().tolist()}")
# Observed (bit-exact across fresh processes): grad0/grad1 both n_big=30, idx=[20..49] (expected all zero)
```

### Key observations (what is ruled out)

1. **Kernel math is correct.** The failing kernel `triton_unk_fused_sum_0` (XBLOCK=1, R0_BLOCK=10, num_warps=2, grid=(40,)) is a standard odometer grouping in ttir: `pid<10 → base=2*pid, trip=2` (IF branch, writes cells 0–19); `pid>=10 → base=pid+10, trip=1` (ELSE branch, writes cells 20–49); store indices `x0 + 5*x1` cover all 50 cells across the 40 programs. Observed: IF branch fully lands, ELSE branch's 30 cells never written ⇒ programs 10–39 did not execute/land.
2. **Second launch of the same binary in the same process is complete** (second backward of the same model, or direct repeated launcher calls: 50/50 written) ⇒ this is a **first-launch** loss, not a per-launch defect.
3. **Follows the binary, not the launcher:** a JIT-compiled (runtime numel signature) kernel of the same source launched through the same C stub writes 50/50; an autotune-winner binary (numel folded to compile-time constants, 13-arg stub) launched through the JIT launcher still writes 30/50. Winner vs JIT launch metadata is field-for-field identical (mix_mode/compile_mode/parallel_mode/coalesce_factor=1/grid/cluster/shared/scratch…); only the binary body and hash differ (npubin same size 3696B, different md5).
4. **Shape-sensitive:** an equivalent kernel whose output has 100 cells ((20,1,5)+(20,5) broadcast) does not lose on first launch — specific grid/grouping configuration dependent.
5. The inductor default backend on the same op does not show this numeric error.

### Expected

A launch of grid=40 should execute all 40 programs and land their stores. Currently, for this kernel form (compile-time-constant-folded numel signature + `scf.if` odometer grouping + SIMT), the first launch silently drops all programs with pid ≥ group_tail (no error, no log); relaunch is normal. Please look at the module-load / first-launch path in triton-ascend (unstructured_in_simt / force_simt_template / AIV mix_mode combination).

### Additional observation (not yet root-caused, for triage)

In the same environment we once saw a process-level `ERR99999 UNKNOWN application exception` crash when a third model's compilation followed in the same process; in nightly runs this family's bool-dtype rows appear as subprocess crashes (`SubprocException`). Possibly the same launch-path family — noting for completeness, not a claim.

**Comment by gavinwang269** [comment]
**Independent reproduction with the same launch-loss signature — but in the forward matmul domain, with three deviations from the characterization in this issue**

Environment: Ascend 910B4, PyTorch 2.14.0, torch_npu 2.14.0+gitb4c7212, CANN 9.1.0, triton-ascend 3.5.0, Inductor backend `triton_experimental` (`TORCHINDUCTOR_NPU_BACKEND=triton_experimental`; no other flags — reproduces with defaults). Every run below used a fresh process with fresh Inductor/Triton caches; results are identical on separate NPU cards.

Minimal repro (3D@1D matmul lowered to a 2D gemv over (25,10)):

```python
import os
os.environ.setdefault("TORCHINDUCTOR_NPU_BACKEND", "triton_experimental")
import torch, torch_npu

torch.manual_seed(0)
A = torch.randn(5, 5, 10, device="npu")
v = torch.randn(10, device="npu")
f = torch.compile(torch.matmul)
out = f(A, v)
ref = torch.matmul(A, v)          # eager
bad = (out != ref).any(dim=-1).nonzero().flatten()
print("wrong rows (of 25):", bad.tolist())          # -> [10..24], stable across runs/cards
print("max_abs dev:", (out - ref).abs().max().item())  # ~8.56 / 10.32 / 10.57 in three runs
```

Shape scan (matrix-vector `(M,K)@(K,)`, plus mirrors; wrong-row set):

| input | wrong rows | note |
|---|---|---|
| (25,8)@(8,) | 8–24 (17/25) | bit-identical across two isolated runs |
| (25,10)@(10,) | 10–24 | boundary = K |
| (25,16)@(16,) | 16–24 | boundary = K |
| (24,10) | 10–23 | boundary = K |
| (40,10) | 10–39 | stale magnitudes differ across processes |
| (41,10) | ~10–40 (+ unstable 11–13) | |
| (11,10) | row 10 only | |
| (10,10), (8,6) | clean | M ≤ K never corrupt |
| `sum` over (25,10) rows | 10–24 | pure row-reduce hits it too |
| (10,)@(5,10,5) rmm | 15/25 cells, stable | 1D@3D mirror |
| true batched 3D mm (3×3×10, 4×4×10, 2×5×10, 5×5×8) | clean | |

Three refinements vs. this issue's model of the defect:

1. **The loss boundary tracks the operand extent, not the odometer group.** Here the first unwritten program is `pid ≥ K` (=10 for K=10), while the generated kernel's odometer `group_tail` is 25. So the lost-launch window is not simply "pid ≥ group_tail".
2. **Persistence across fresh launches:** the issue states the same binary relaunches complete. For this forward-matmul kernel, fresh processes repeatedly leave the same *cell set* unwritten (rows ≥ K); only the stale *values* move (they look like leftover buffer content). That is consistent with "cells never written" rather than a race.
3. **Domain:** this is forward `matmul`/gemv and a plain row-reduction under the `triton_experimental` Inductor backend — not broadcast-backward.

We line-checked the generated Triton kernel: the math and the index mapping are correct for every program id, including the unwritten rows — pointing at the launcher's program distribution rather than codegen. Related GitCode-side tracker for the broadcast-backward odometer family: Ascend/pytorch#4952.

**Comment by hongziqi** [comment]
Thanks for your feedback. We will arrange relevant engineers to investigate this issue later.
